### PR TITLE
Bye, bye nokogiri! Use aws-sdk instead (WIP

### PR DESCRIPTION
**Pull request by pburkholder**
On March 30 aws-sdk v2 changed over to sax processing so the gem can be used just with rexml, no need for Nokogiri or other compiled DOM processing[0].

Since the build-essentials hell of compile time fog installation has been one of the biggest frustrations in using this cookbook, this PR uses the Aws::Route53 client instead. 

At this point it seems to work but testing has been minimal, awaiting comments from mainteners. Also, I've removed `alias` support until we get vanilla records worked out.

Also:

* Remove test-kitchen cruft in spec/ dir
* Enable basic rspec/chefspec test to ensure pre-intergraion sanity
* Add Berksfile for chefdk compatibility
* Remove unneeded reference to session token

Ping: @scalp42 @bookest - Hope this helps!
@cwebberops As discussed the other day.

### CODE REVIEW

**Comment by slyness** [comment]
As it seems we're always fighting fog and nokogiri I like what you're doing here. I'll find some time to review and comment as soon as possible.

**Comment by josacar** [comment]
I'd be great to have it feature-complete, I'll try to give it a try and help

**Comment by pburkholder** [comment]
@josacar If you could add alias support that would be great. I'll be using this branch for some of my own work over the next couple of weeks, which should expose any superficial bugs.

**Comment by scalp42** [comment]
@pburkholder good stuff :+1:

File: recipes/default.rb [hunk 1]
```diff
@@ -17,11 +17,9 @@
 # limitations under the License.
 #
 
-include_recipe 'xml::ruby'
-
-chef_gem "fog" do
+chef_gem 'aws-sdk' do
   action :install
-  version node['route53']['fog_version']
+  compile_time false if Chef::Resource::ChefGem.method_defined?(:compile_time)
```
Comment (scalp42) [inline recipes/default.rb]: I don't think it'll work on all versions of Chef, see: https://github.com/opscode-cookbooks/aws/pull/110

Prefer:
```ruby
compile_time true if Chef::Resource::ChefGem.instance_methods(false).
                                                                            include?(:compile_time)
```

**Comment by pburkholder** [comment]
@scalp42 Thanks for catching that

**Comment by pburkholder** [comment]
Forgot to leave this reference: https://github.com/aws/aws-sdk-ruby/pull/745

**Comment by scalp42** [comment]
Any chance to see this merged @slyness ?

**Comment by till** [comment]
@pburkholder are you using this currently? Or would you mind updating where this PR is stuck (if it is)?

**Comment by pburkholder** [comment]
I'm behind on my various half-completed PRs. I wrote this to fill a need, which it does, and now I do need to tie a bow on it. May over the holidays.....(or sooner).

**Comment by till** [comment]
@pburkholder if it helps, I could send you a PR to address open items. Let me know your thoughts.

**Comment by pburkholder** [comment]
That would be great. I can merge them into this PR and then pester the heavy water folks to merge into master.

**Comment by till** [comment]
@pburkholder so, can you share what's missing? :)

**Comment by nadirollo** [comment]
Hi! I was on a mission to get rid of build-essentials from my runs and my last piece was the dependency of this one!!!
Is there something missing in order to merge this PR that we can help with?

**Comment by till** [comment]
Just as a heads up, we wrote our own provider which wraps around a simple library which in turn uses the `aws-sdk` (v1 currently, for other dependencies in our stacks).

I can spin this off from the our cookbook repo, or you can _steal_ it ;))
https://github.com/till/easybib-cookbooks/tree/master/ies-route53

**Comment by scalp42** [comment]
heads up until it gets merged https://github.com/scalp42/route53/tree/pdb for https://github.com/cwebberOps/route53/pull/1

**Comment by slyness** [comment]
I think this is a good idea. I've been (finally) taking a look at the rest of the current PRs for this cookbook and I think it might be a good idea to cut at least one more minor release using the existing fog configuration and the follow that up with a major version release moving to aws-sdk. I'll start working on making that happen.

**Comment by slyness** [comment]
@pburkholder can you please now rebase this PR again.

 * Please remove updates to `metadata.rb` and `CHANGELOG.md` I'll handle that when cutting a new release. These changes will be released in a major 1.x release.

* It is also preferred to not include a Berksfile for this repository as it breaks the existing test-kitchen suites for the pre-existing use of librarian and the Cheffile. It's fine to keep a Berksfile in your local environment. We now exclude it for the repository in `.gitignore`

* also, note that there is some new support for geo_location in the 0.4.2 release so you might might need to add that to your update. 

Thanks again.

**Comment by scalp42** [comment]
Please make sure to merge https://github.com/cwebberOps/route53/pull/1 :warning:

**Comment by pburkholder** [comment]
@scalp42 this is in progress. There are more conflicts than I expected so it may be a few more days until I have time to work it again.

**Comment by slyness** [comment]
@scalp42 sounds good. no problem here.

**Comment by pburkholder** [comment]
This got backlogged and now I'm out for a few weeks. Will pick up on in April.

**Comment by slyness** [comment]
No worries. Thanks for the update.

**Comment by pburkholder** [comment]
@scalp42 @slyness This has all the bits for geolocation and alias_record back in, and the current tests pass kitchen and rspec.

I still need to test for real in EC2. I should be able to do so yet this week.

**Comment by tas50** [comment]
@pburkholder If you're still interested in getting this merged in I'd love to simplify things and prep this cookbook for the eventual merge into the aws cookbook.

**Comment by pburkholder** [comment]
@tas50 I am interested. I had it mergeable a few weeks ago, and had done some testing, but not enough to get around to updating this PR.

How would you like to proceed?

**Comment by blairham** [comment]
+1

**Comment by tas50** [comment]
This has been manually merged in.

**Pull request closed without merging**